src/works/xbeni/components/reporte-pe/reporte-pe.js

```diff
@@ -0,0 +1,346 @@
+import { LitElement, html } from 'lit-element';
+import { RpeStyles } from '../../archivos_comunes/ac_reportePe/styles';
+import { loadEmpleadosRpe, getDatosReporteRpe } from '../../archivos_comunes/ac_reportePe/mocks';
+import { CONSTANTS_RPE } from '../../archivos_comunes/ac_reportePe/constantes';
+import {
+  svgXBeniRpeOrderString,
+  svgBeniRpeOrdenarInt,
+  svgBeniRpeIconRight,
+  svgBeniRpeIconLeft,
+} from '../../archivos_comunes/ac_reportePe/svg_icons';
+import { nothing } from 'lit-html';
+
+class BeniReportePermisosEmpleado extends LitElement {
+  constructor() {
+    super();
+    this.tituloReporte = 'Reporte de permisos detallado';
+    this.empleadosRpe = loadEmpleadosRpe();
+    this.datosReporteRpe = [];
+    this.index = 0;
+    this.from = 0;
+    this.nElements = 10;
+    this.stepper = [];
+  }
+
+  static get properties() {
+    return {
+      tituloReporte: { type: String },
+      empleadosRpe: { type: Object },
+      datosReporteRpe: { type: Array },
+      controlGenerarReporte: { type: Boolean, Attribute: false },
+      index: { type: Number, attribute: false },
+      from: { type: Number, attribute: false },
+      nElements: { type: Number },
+      stepper: { type: Array, attribute: false },
+    };
+  }
+
+  static get styles() {
+    return [RpeStyles];
+  }
+
+  render() {
+    return html`
+      <div class="divBodyRpe">
+        <div class="divBodyControlRpe">
+          <div class="divHeaderControlRpe">
+            <label>${this.tituloReporte}</label>
+          </div>
+    
+          <div class="divMainControlRpe">
+            <div>
+              <div class="divFlexRpe">
+                <div class="divCamposRpe">
+                  <label>Empleado:</label>
+                </div>
+                <div class="divCamposDatos">
+                  <select name="empleadosRpe" id="${CONSTANTS_RPE.idSelectEmpleadosRpe}" class="selectRpe">
+                    <option value="-1">Selecciona un empleado</option>
+                      ${Object.keys(this.empleadosRpe).map(
+                        (item) => html`
+                          <option value="${this.empleadosRpe[item].id_empleado}">
+                            ${this.empleadosRpe[item].nombre}
+                          </option>
+                        `,
+                      )}
+                  </select>
+                </div>
+              </div>
+    
+              <div class="divFlexRpe">
+                <div class="divCamposRpe">
+                  <label>Fecha de inicio:</label>
+                </div>
+                <div class="divCamposDatos">
+                  <input type="date" id="${CONSTANTS_RPE.idFechaInicioRpe}" class="inputFechasRpe"/>
```

Comment:
No es una práctica recomendable sacar los identificadores html a una constante y menos a un fichero a parte ya que al ser "identificadores", son únicos y no se van a modificar, y más cuando hablamos de web-components donde cada componente está encapsulado y sus referencias no van a entrar en conflicto con las de otros componentes.

Tal como está ahora, para ver el valor de un identificador o de un querySelector me tengo que ir a otro archivo, perdiendo el foco del código, cuando realmente no aporta nada..

Los identificadores ya sean en html o en un querySelector() deben de definirse en el propio código para facilitar la lectura del mismo.

Reply:
A que te refieres con esto? Que los elementos que esten definidos en constantes <select name="empleadosRpe" id="${CONSTANTS_RPE.idSelectEmpleadosRpe}"> los cambie a esto <select name="empleadosRpe" id="idSelectEmpleadosRpe"> o que los defina directamente en el propio componente 
  static get properties() {
    return {
      idSelectEmpleado: { type: String, attribute: false },
    }
  constructor() {
    super();
    this.idSelectEmpleado = 'idSelectEmpleadosRpe'
   }

cual de esas dos formas debo aplicar??

Reply:
Código refactorizado